File: Taller Tipos de clases/C#/Clase Interfaz/Operacion.cs
interface IOperacion
{
    int Ejecutar(int a, int b);
}

class Suma : IOperacion
{
    public int BaseValue;

    public Suma()
    {
        BaseValue = 0;
    }

    public Suma(int baseValue)
    {
        BaseValue = baseValue;
    }

    public Suma(Suma s)
    {
        BaseValue = s.BaseValue;
    }

    public int Ejecutar(int a, int b)
    {
        return a + b + BaseValue;
    }
}

File: Taller Tipos de clases/Go/Clase Interfaz/Operacion.go
interface Operacion {
    int ejecutar(int a, int b);
}

class Suma implements Operacion {
    int base;

    public Suma() {
        base = 0;
    }

    public Suma(int base) {
        this.base = base;
    }

    public Suma(Suma s) {
        this.base = s.base;
    }

    public int ejecutar(int a, int b) {
        return a + b + base;
    }
}

